src/lib.rs

```diff
@@ -2035,7 +2042,14 @@ impl Build {
             self.assemble_progressive(dst, chunk)?;
         }
 
-        if self.cuda {
+        if self.cuda
```

Comment:
Since you have this check (are we cuda and have \>1 .cu file), it's probably worth pulling out into a function.

Reply:
Nevermind, this is different.

Reply:
As in `if self.cuda && self.cuda_files_count() > X` or as in `if self.cuda_files_count() > X`?

Reply:
Well, a cuda_files_count() would make a sense, because the formatting is unbearable :-)

Reply:
Yeah, that would be reasonable.